### DISCUSSION TITLE
Doubt about group constant file suffix

### DISCUSSION THREAD

**Discussion by js-jixu** [arfc/moltres#200] [Q&A Reactor multiphysics] [answered]
Hi, everyone.

I want to do a IAEA2D benchmark by Moltres, which needs to modify group constants. Then  I want to correspond suffix of files of group constants to their physical meaning. Now, I've determined some suffixes but still had some undetermined ones.

I want to know the meanings of those group constant file suffix which aren't found in MoltresPhysicalKernels: FISSE, FLUX, NUBAR, CHI_T.

Any answers are deeply apperciated!

**Comment by smpark7** [comment, marked as answer]
FISSE = Energy produced per fission (~200MeV), usually represented with greek letters epsilon (ε) or kappa (κ)

FLUX = Group flux integrated over the whole domain. Moltres does not use this in any simulations. We'll probably remove this in future.

NUBAR = Average neutrons produced per fission, represented with the greek letter nu (ν) with a bar (straight line) above it to signify that it is an average value. I think Moltres does not use this either because it uses NSF (nu sigma_f) instead.

CHI_T = Total fission probability spectrum (prompt + delayed) represented with the greek letter (χ)

**Reply by js-jixu** [reply]
Hi, Sun Myung. 

Thanks a lot for your reply. 

I‘m writing a IAEA2D benchmark input file. My understanding of this benchmark is to solve a two-group steady-state neutron diffusion equation for four materials and the group constants for the materials are given. First, given a hypothetical eigenvalue, the neutron flux is obtained by solving the diffusion equations. Then use the neutron flux calculated in the previous step to correct the new eigenvalues. This iteration goes on to get a final eigenvalue. Should I use CoupledFissionEigenKernel and a postprocessor provided to CoupledFissionEigenKernel? Can I find any reference from Moltres's prbolems floder?

With best wishes,

Xu.

**Reply by smpark7** [reply]
The `moltres/problems/2021-cnrs-benchmark/phase-0/nts.i` input file should give you a good idea of how to set up your input file. It involves a six-group neutronics eigenvalue problem in a 200cm-by-200cm static molten salt domain. The [`InversePowerMethod`](https://mooseframework.inl.gov/source/executioners/InversePowerMethod.html) executioner is suitable for calculating the multiplication factor eigenvalue. Note that this input file uses the `[Nt]` action to automatically initialize the required variables, kernels, and BCs for the six-group neutron diffusion equation.

Since the IAEA2D benchmark does not model precursors, you'll want to disable precursors by removing the `[Precursors]` section and setting `account_delayed=false` in `[GlobalParams]`. Moltres will use CHI_T for the fission spectrum χ1 and χ2. However, Moltres will still require you to provide CHI_P values by default. You can set CHI_P to any arbitrary value to avoid getting an error.

**Reply by js-jixu** [reply]
Thank you, Sun Myung. 

I wrote a new input file today referring to the input file you mentioned, but I encountered some problems with the results. I put it under a new discussion so that our discussion can be referenced by later beginners like me.

**Reply by js-jixu** [reply]
Hi, Sun Myung.

I wonder if Moltres will use `CHIP` when `sss2_input=1` ， and `CHIT` can be given arbitrarily (because Moltres does not use it for computation). Correspondingly, when `sss2_input=0` ， Moltres will use `CHI_T` , and `CHI_P` can be given arbitrarily. Can I understand that if I use OpenMC to calculate the neutron cross-section, then I need to provide `CHIP` for the calculation, and `CHIT` to prevent Moltres from reporting an error (in fact, `CHIT` does not participate in the calculation); if I use Serpent2 to calculate the neutron cross-section, I need to provide `CHI_T` used for calculation, `CHI_P` used to prevent Moltres from reporting errors (in fact, `CHI_P` does not participate in calculation)? Is this understanding correct?

I would also like to ask you about the role of the scaling parameter. In this case, does the unit of neutron flux change to `10^3/(cm^2·s)` when you use `scaling=1e3` ?

![scaling](https://user-images.githubusercontent.com/99813361/168523593-b113b813-8208-43a7-8df0-f4d3ba302b53.png)

**Reply by smpark7** [reply]
Hi Xu. That's not correct. Whether Moltres uses `CHI_T` or `CHI_P` depends on the `account_delayed` parameter. If `account_delayed=true`, then Moltres uses both `CHI_P` and `CHI_D` for prompt and delayed neutrons, respectively. Otherwise, Moltres uses `CHI_T` which is the flux-averaged total. `CHI_T` is usually sufficient for steady-state calculations.

`sss2_input` determines how Moltres reads the multigroup scattering cross sections. This originated from differences in SCALE/NEWT and Serpent 2 output files. SCALE/NEWT provides scattering cross sections ordered by the in-scattering group, i.e. 1->1, 2->1, 3->1, ..., n->1, 1->2, ..., n->n while Serpent 2 provides scattering cross sections ordered by the out-scattering group, i.e. 1->1, 1->2, 1->3, ..., 1->n, 2->1, ..., n->n. As you may have guessed, the parameter is named after `sss2`, the name of the Serpent 2 executable. OpenMC scattering cross sections follow the same format as Serpent 2 as long as you follow the steps outlined in the `tutorial/godiva` tutorial. So you'll want to leave `sss2_input=true` if you use OpenMC group constants.

**Reply by js-jixu** [reply]
Thanks a lot, Sun Myung. Your reply solved my doubts about the Moltres group constant suffix
